codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerOperationRegistryGenerator.kt

```diff
@@ -23,9 +23,9 @@ import software.amazon.smithy.rust.codegen.core.rustlang.withBlockTemplate
 import software.amazon.smithy.rust.codegen.core.rustlang.writable
 import software.amazon.smithy.rust.codegen.core.smithy.CodegenContext
 import software.amazon.smithy.rust.codegen.core.smithy.CodegenTarget
-import software.amazon.smithy.rust.codegen.core.smithy.Errors
-import software.amazon.smithy.rust.codegen.core.smithy.Inputs
-import software.amazon.smithy.rust.codegen.core.smithy.Outputs
+import software.amazon.smithy.rust.codegen.core.smithy.ErrorsModule
+import software.amazon.smithy.rust.codegen.core.smithy.InputsModule
+import software.amazon.smithy.rust.codegen.core.smithy.OutputsModule
```

Comment:
I see these referenced, but I don't see them defined anywhere in the PR. Did you miss a file?

It would be really cool if we could represent the module hierarchy in Kotlin. Maybe something like this:
```kotlin
// Have one of these in codegen-client, and another one in codegen-server
object Modules {
    val Errors = RustModule.newModule(/* ... */)
    object Models {
        val Paginators = RustModule.newModule(/* ... */)
        val Inputs = RustModule.newModule(/* ... */)
        val Outputs = RustModule.newModule(/* ... */)
    }
}
```
Then when referencing it:
```kotlin
writer.withModule(Modules.Models.Inputs) { /* ... */ }
```

Although, it kind of breaks down if you want to put things directly into `Models` in the example above... needs some workshopping. `object`s in Kotlin can inherit things, so that might be the trick. Have a `ToModule` interface or something that it can inherit:

```kotlin
interface ToModule {
    fun toModule(): RustModule
}

object Modules {
    val Errors = RustModule.newModule(/* ... */)
    object Models : ToModule {
        val Self = RustModule.newModule(/* ... */)
        override fun toModule(): RustModule = Self

        val Paginators = RustModule.newModule(/* ... */)
        val Inputs = RustModule.newModule(/* ... */)
        val Outputs = RustModule.newModule(/* ... */)
    }
}

Reply:
they aren't added (they already existed). I just started using them directly instead of SourceLocation

Reply:
agreed this would be cool, going to punt at that for now though